docs/troubleshooting.md

```diff
@@ -50,6 +50,28 @@ For example, in [Posit Connect 2024.12.0](https://docs.posit.co/connect/news/#po
 Gradio app support was introduced. When deploying a Gradio app to a server
 running an older version of Posit Connect than 2024.12.0, it will error.
 
+### `renv` errors on deployment
+
+When deploying projects using `R`, it is indispensable for the Posit Publisher extension
+to have access to a record of the dependencies used by your project.
+This record is expected to be found as a `lockfile` (commonly named as `renv.lock`) generated by `renv`,
+since the [`renv` package](https://rstudio.github.io/renv/articles/renv.html)
+is a great choice to create reproducible environments.
```

Comment:
I found this section a bit awkwardly worded, and sort of out of the order I'd expect the explanation.

```suggestion
When deploying `R` projects, it is essential that Posit Publisher has
access to a record of the dependencies used by your project to reproduce
the project on Posit Connect.
The [`renv` package](https://rstudio.github.io/renv/articles/renv.html) can
create that record as a `lockfile` (commonly named `renv.lock`).
```